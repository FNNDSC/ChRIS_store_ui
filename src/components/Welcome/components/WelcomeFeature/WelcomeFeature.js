import React from 'react';
import PropTypes from 'prop-types';
import './WelcomeFeature.css';

const WelcomeFeature = props => (
  <div className="welcome-feature">
    <img src={props.img} alt="" className="welcome-feature-img" />
    <div className="welcome-feature-text">
      {props.name}
    </div>
  </div>
);

WelcomeFeature.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default WelcomeFeature;
