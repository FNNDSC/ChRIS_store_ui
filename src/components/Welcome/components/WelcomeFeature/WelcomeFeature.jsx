import React from 'react';
import PropTypes from 'prop-types';
import styles from './WelcomeFeature.module.css';

const WelcomeFeature = ({
  img, name, url,
}) => (
  <div>
    <div className={styles.welcomeFeature}>
      <a href={url}>
        <img src={img} alt={name} className={styles.welcomeFeatureImg} />
      </a>
      <div className={styles.welcomeFeatureText}>{name}</div>
    </div>
  </div>
);

WelcomeFeature.defaultProps = {
  url: PropTypes.string,
};

WelcomeFeature.propTypes = {
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  url: PropTypes.string,
};

export default WelcomeFeature;
