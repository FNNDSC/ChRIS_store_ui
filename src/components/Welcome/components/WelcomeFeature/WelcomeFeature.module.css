.welcome-feature {
  /* display */
  display: inline-flex;
  padding: 0.4em;
}

.welcome-feature-img {
  /* display */
  width: 4em;
  height: 4em;

  /* flex */
  flex-shrink: 0;
}

.welcome-feature-text {
  /* color */
  color: #fafafa;
  color: var(--pf-black-100);

  /* font */
  font-size: 1.2em;
  font-weight: 500;
  line-height: 2;

  /* display */
  padding: 0.65em;
}

/* ============================== */
/* ---------- RESPONSIVE -------- */
/* ============================== */

@media (min-width: 480px) {
  .welcome-feature {
    /* display */
    padding: 0.5em;
  }

  .welcome-feature-img {
    /* display */
    width: 4.5em;
    height: 4.5em;
  }

  .welcome-feature-text {
    /* font */
    font-size: 1.3em;
    line-height: 2.5;
  }
}

@media (min-width: 768px)  {
  .welcome-feature {
    /* display */
    padding: 1em;

    /* flex */
    flex-direction: column;
  }

  .welcome-feature-img {
    /* display */
    display: block;
    width: 5em;
    height: 5em;
    margin: 0 auto;
  }

  .welcome-feature-text {
    /* font */
    font-size: 1em;
    text-align: center;
    line-height: 1.3;
  }
}
