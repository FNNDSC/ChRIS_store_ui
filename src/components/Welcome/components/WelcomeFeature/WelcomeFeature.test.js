import React from 'react';
import { shallow, mount } from 'enzyme';
import WelcomeFeature from './WelcomeFeature';

describe('WelcomeFeature', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<WelcomeFeature img="" name="" />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render welcome-feature div', () => {
    expect(wrapper.find('div.welcome-feature')).toHaveLength(1);
  });

  it('should render welcome-feature-img img', () => {
    expect(wrapper.find('img.welcome-feature-img')).toHaveLength(1);
  });

  it('should render welcome-feature-text div', () => {
    expect(wrapper.find('div.welcome-feature-text')).toHaveLength(1);
  });
});

describe('mounted WelcomeFeature', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<WelcomeFeature img="" name="" />);
  });

  it('renders the value of img', () => {
    wrapper.setProps({ img: 'testImageSrc' });
    expect(wrapper.find('img').prop('src')).toEqual('testImageSrc');
  });

  it('renders the value of name', () => {
    wrapper.setProps({ name: 'testName' });
    expect(wrapper.find('div.welcome-feature-text').text()).toEqual('testName');
  });
});
