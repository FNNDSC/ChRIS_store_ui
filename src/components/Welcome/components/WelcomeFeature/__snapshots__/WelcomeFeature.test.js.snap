// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeFeature should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WelcomeFeature
    img=""
    name=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          className="welcome-feature-img"
          src=""
/>,
        <div
          className="welcome-feature-text"
/>,
      ],
      "className": "welcome-feature",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "welcome-feature-img",
          "src": "",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "welcome-feature-text",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className="welcome-feature-img"
            src=""
/>,
          <div
            className="welcome-feature-text"
/>,
        ],
        "className": "welcome-feature",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "welcome-feature-img",
            "src": "",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "welcome-feature-text",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
